### PR TITLE
Fix AutoDelta with deterministic sites

### LINKED ISSUE #945: AutoDelta raises duplication error
Currently, using AutoDelta on some models generates sort of `AssertionError: all sites must have unique names but got `sd_dip_rnde` duplicated` error. More information to come...

### PR DESCRIPTION
**Pull request by fehiepsi**
Fixes #945 by adding `handlers.block()` in `setup_prototype`. When there are deterministic sites, the `postprocess_fn` traces the model (we can skip tracing in `postprocess_fn` but it is beyond the scope of this PR), hence creates duplicated site during `svi.init(...)`.

Also added small fixes for some warnings in CI.

### CODE REVIEW

File: docs/source/distributions.rst [hunk 1]
```diff
@@ -638,14 +638,6 @@ ExpTransform
...
     :member-order: bysource
     
-SoftplusTransform
------------------
-.. autoclass:: numpyro.distributions.transforms.SoftplusTransform
-    :members:
-    :undoc-members:
-    :show-inheritance:
-    :member-order: bysource
-
```
Comment (fehiepsi) [inline docs/source/distributions.rst:648]: this entry is duplicated

File: examples/neutra.py [hunk 2]
```diff
@@ -116,28 +116,28 @@ def main(args):
     ax1.set_title('Autoguide training loss\n(after 1000 steps)')
 
     ax2.contourf(X1, X2, P, cmap='OrRd')
-    sns.kdeplot(guide_samples[:, 0], guide_samples[:, 1], n_levels=30, ax=ax2)
+    sns.kdeplot(x=guide_samples[:, 0], y=guide_samples[:, 1], n_levels=30, ax=ax2)
```
Comment (fehiepsi) [inline examples/neutra.py:119]: seaborn raises FutureWarning to specify explicitly keyword arguments

File: test/pyroapi/test_pyroapi.py [hunk 3]
```diff
@@ -8,7 +8,7 @@
 pytestmark = pytest.mark.filterwarnings("ignore::numpyro.compat.util.UnsupportedAPIWarning")
 
 
-@pytest.yield_fixture
+@pytest.fixture
```
Comment (fehiepsi) [inline test/pyroapi/test_pyroapi.py:11]: both are the same, but the former is deprecated

File: numpyro/infer/autoguide.py [hunk 4]
```diff
@@ -287,15 +287,15 @@ class AutoDelta(AutoGuide):
...
         super().__init__(model, prefix=prefix, init_loc_fn=init_loc_fn, create_plates=create_plates)
 
     def _setup_prototype(self, *args, **kwargs):
         super()._setup_prototype(*args, **kwargs)
-        self._init_locs = {
-            k: v for k, v in self._postprocess_fn(self._init_locs).items() if k in self._init_locs
-        }
+        with numpyro.handlers.block():
+            self._init_locs = {
+                k: v for k, v in self._postprocess_fn(self._init_locs).items() if k in self._init_locs
```
Comment (fritzo) [inline numpyro/infer/autoguide.py:297]: Looks reasonable. What does `self._postprocess_fn()` do? Should it always be blocked, or only here? I guess an alternative place to block would be in `.__call__()`:
```py
def __call__(self, *args, **kwargs):
    if self.prototype_trace is None:
        with numpyro.handlers.block():
            self._setup_prototype(*args, **kwargs)
```
then you could remove the `block` from `_setup_prototype`...

**Review by fritzo — APPROVED** [review]
LGTM

File: numpyro/infer/autoguide.py (hunk 4 above)
Comment (fehiepsi) [reply numpyro/infer/autoguide.py:297]: > What does self._postprocess_fn() do? Should it always be blocked, or only here?

Only here I think. `postprocess_fn` transforms unconstrained values to constrained values and also collects deterministic sites.

> I guess an alternative place to block would be in .__call__():

You are right. Do you want to move `block` there? Both places look good to me.

Comment (fritzo) [reply numpyro/infer/autoguide.py:297]: Your original design looks good.

**Pull request merged**